src/MassTransit/DependencyInjection/Configuration/ServiceCollectionBusConfigurator.cs

```diff
@@ -191,13 +199,19 @@ public void AddRider(Action<IRiderRegistrationConfigurator<TBus>> configure)
             configure?.Invoke(configurator);
         }
 
+        public override void AddConfigureEndpointsCallback(ConfigureEndpointsCallback callback)
+        {
+            this.AddSingleton(provider => Bind<TBus>.Create((IConfigureReceiveEndpoint)new ConfigureReceiveEndpointDelegate(callback)));
+        }
+
         public override void AddConfigureEndpointsCallback(ConfigureEndpointsProviderCallback callback)
         {
             if (callback == null)
                 throw new ArgumentNullException(nameof(callback));
 
-            this.TryAddSingleton<IConfigureReceiveEndpoint>(provider =>
-                new ConfigureReceiveEndpointDelegateProvider(provider.GetRequiredService<Bind<TBus, IBusRegistrationContext>>().Value, callback));
+            this.AddSingleton(provider => Bind<TBus>.Create((IConfigureReceiveEndpoint)new ConfigureReceiveEndpointDelegateProvider(
```

Comment:
With this change, these are no longer added as singletons, they should allow multiple configuration callbacks to be added.

Reply:
that's exactly what `AddSingleton` is used (instead of `TryAddSingleton `)